fdbclient/ClientLibManagement.actor.cpp

```diff
@@ -123,7 +124,22 @@ ClientLibChecksumAlg getChecksumAlgByName(std::string_view checksumAlgName) {
 namespace {
 
 bool isValidTargetStatus(ClientLibStatus status) {
-	return status == ClientLibStatus::AVAILABLE || status == ClientLibStatus::DISABLED;
+	return status == ClientLibStatus::AVAILABLE || status == ClientLibStatus::DISABLED ||
+	       status == ClientLibStatus::DOWNLOAD || status == ClientLibStatus::ACTIVE;
+}
+
+bool isAvailableForDownload(ClientLibStatus status) {
+	return status == ClientLibStatus::AVAILABLE || status == ClientLibStatus::DOWNLOAD ||
+	       status == ClientLibStatus::ACTIVE;
+}
+
+void updateClientLibChangeCounter(Transaction& tr, ClientLibStatus status) {
+	static const int64_t counterIncVal = 1;
+	if (status == ClientLibStatus::DOWNLOAD || status == ClientLibStatus::ACTIVE) {
```

Comment:
Are there states that would trigger a client to unload a library (e.g. if we switched from `ACTIVE/DOWNLOAD` to `DISABLED`)?

Reply:
A good point. We should also notify about libraries losing "ACTIVE/DOWNLOAD" status or libraries with "ACTIVE/DOWNLOAD" being deleted.